### PR TITLE
Python code examples for file IO in xml and yml format

### LINKED ISSUE #14665: File Input Output with XML and YML tutorial does not have Python code examples
The File Input Output with XML and YML tutorial (at https://docs.opencv.org/4.1.0/dd/d74/tutorial_file_input_output_with_xml_yml.html) caters to C++ programmers but not Python programmers. It would be useful if there was a Python tutorial for this. Since the C++ tutorial is version-independent, so should the corresponding Python tutorial.

### PR DESCRIPTION
**Pull request by ihsan314**
resolves #14665 

### This pullrequest changes
the fact that Python tutorials do not exist for File Input Output with XML and YML.

```
force_builders=Docs
```

### CODE REVIEW

**Comment by alalek** [comment]
Thank you for contribution!

Please add fixes to make "Docs" builder [green](http://pullrequest.opencv.org/buildbot/builders/precommit_docs/builds/20588).

**Comment by asmorkalov** [comment]
@alalek Docs build is green. Could you look at the patch.

**Comment by alalek** [comment]
This patch should [go](https://github.com/opencv/opencv/wiki/How_to_contribute#faq) into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first. We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

So, please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no needs to re-open PR, apply changes "inplace".

File: samples/python/tutorial_code/core/file_input_output/file_input_output.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+import cv2 as cv
+import numpy as np
```
Comment (alalek) [inline samples/python/tutorial_code/core/file_input_output/file_input_output.py]: It is better to import numpy before cv2 (we have reports where this order of imports doesn't work)

File: samples/python/tutorial_code/core/file_input_output/file_input_output.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import sys
+
+def help(filename):
+    print (filename, 'shows the usage of the OpenCV serialization functionality.',
```
Comment (alalek) [inline samples/python/tutorial_code/core/file_input_output/file_input_output.py]: Please use normal print from Python3 (+ `from __future__ import print_function` for Python 2.7).

Also it is better to use docstring.

**Comment by ihsan314** [comment]
Thanks for reviewing the pull request. I will work on all the changes by the end of this week.

**Comment by asmorkalov** [comment]
@ihsan314 Do you have any progress on the patch?

**Comment by ihsan314** [comment]
I didn't have that much time in the last two weeks, so here they are today. The docstrings will be added hopefully by tomorrow.

**Comment by asmorkalov** [comment]
@ihsan314 Please pay attention on CI status, it still reports some issues with the code.

**Comment by ihsan314** [comment]
The CI status had only one failed build, and that failed build was on Mac. I fast-forwarded my branch and tried the failed build on my machine. It worked on mine, so I hope it will pass. I will change the strings to docstrings next.

**Comment by asmorkalov** [comment]
@ihsan314 Please remove "WIP" from the PR description to trigger CI builds automatically.

**Comment by asmorkalov** [comment]
@alalek could you look at it again.

File: doc/tutorials/core/file_input_output_with_xml_yml/file_input_output_with_xml_yml.markdown [hunk 3]
```diff
@@ -26,102 +26,122 @@ library.
 Here's a sample code of how to achieve all the stuff enumerated at the goal list.
 
 @include cpp/tutorial_code/core/file_input_output/file_input_output.cpp
+@end_toggle
+
+@add_toggle_python
+You can [download this from here
+](https://github.com/opencv/opencv/tree/master/samples/python/tutorial_code/core/file_input_output/file_input_output.py) or find it in the
```
Comment (alalek) [inline doc/tutorials/core/file_input_output_with_xml_yml/file_input_output_with_xml_yml.markdown]: > master

Please use "3.4" in URLs (as patch is targeted to 3.4 branch).

**Pull request merged**